raiden-ts/src/transport/epics.ts

```diff
@@ -1503,3 +1521,398 @@ export const deliveredEpic = (
     }),
   );
 };
+
+type CallInfo = { callId: string; peerId: string; peerAddress: Address };
+
+// fetches and caches matrix set turnServer
+const _matrixIceServersCache = new WeakMap<MatrixClient, [number, RTCIceServer[]]>();
+async function getMatrixIceServers(matrix: MatrixClient): Promise<RTCIceServer[]> {
+  const cached = _matrixIceServersCache.get(matrix);
+  if (cached && Date.now() < cached[0]) return cached[1];
+  const fetched = ((await matrix.turnServer()) as unknown) as
```

Comment:
Can't we use `matrix.getTurnServers()` from [here](https://matrix-org.github.io/matrix-js-sdk/5.2.0/module-client.MatrixClient.html#getTurnServers)? At least the signature looks much easier and I don't understand how this works here because we don't use the `matrix.turnServer(callback)`

Reply:
No, unfortunately, because I had to disable matrix-js-sdk's handling of the `m.call.*` events, or it interferes too much in our approach, and disabling it also disables the repeating function  which fetches and populates `getTurnServers` (it's cached internally).
My version here works by using a cache to store previous `turnServer()` response (which is the actual request made) up to the `ttl` of it, or 1h for empty/failing responses, parses it in the format of `RTCIceServer` interface, and returns the cached version of it, if so.